lib/puppet/parser/functions/prefix.rb

```diff
@@ -10,6 +10,11 @@ module Puppet::Parser::Functions
         prefix(['a','b','c'], 'p')
 
     Will return: ['pa','pb','pc']
+
+    Note that since Puppet 4.0.0 the general way to modify values is in array is by using the map
```

Comment:
the line break doesn't need to be there and The 'p' in Puppet should be uppercase.